### PR TITLE
Feature/ipad polishing

### PR DESCRIPTION
**Pull request by denis-ryzhkov-makeen**

### CODE REVIEW

File: LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift [hunk 1]
```diff
@@ -11,8 +11,7 @@ final class AboutVC: UIViewController {
     
     private var screenTitleLabel: UILabel = {
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift:12]: create a custom view for titleLabel and reuse it everywhere

File: LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift [hunk 2]
```diff
@@ -40,9 +39,9 @@ final class AboutVC: UIViewController {
         let isPad = UIDevice.current.isPad
         if isPad {
             view.addSubview(screenTitleLabel)
-            screenTitleLabel.snp.makeConstraints { make in
-                make.top.equalTo(view.safeAreaLayoutGuide).offset(24)
-                make.leading.equalToSuperview().offset(24)
+            screenTitleLabel.snp.makeConstraints {
+                $0.top.equalTo(view.safeAreaLayoutGuide)
+                $0.leading.equalToSuperview().offset(40)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift:44]: move to constants

File: LocationServices/LocationServices/Scenes/About/Sub-Scenes/Attribution/Controller/AttributionVC.swift [hunk 3]
```diff
@@ -141,15 +141,20 @@ final class AttributionVC: UIViewController {
...
             $0.leading.trailing.equalToSuperview()
             $0.height.equalTo(1)
         }
         
         partnerAttributionTitleLabel.snp.makeConstraints {
-            $0.top.equalTo(separatorView.snp.bottom).offset(24)
+            if isIpad {
+                $0.top.equalTo(view.safeAreaLayoutGuide)
+            } else {
+                $0.top.equalTo(separatorView.snp.bottom).offset(24)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/About/Sub-Scenes/Attribution/Controller/AttributionVC.swift:156]: move to constants

File: LocationServices/LocationServices/Scenes/Explore/Controller/ExploreVC.swift [hunk 4]
```diff
@@ -378,6 +367,15 @@ private extension ExploreVC {
         changeSeachBarVisibility(isHidden: isVisible)
     }
     
+    @objc private func exploreActionButtonsVisibilityChanged(_ notification: Notification) {
+        if let geofenceIsHidden = notification.userInfo?["geofenceIsHidden"] as? Bool {
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Explore/Controller/ExploreVC.swift:371]: move to constants

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Controller/DirectionVC.swift [hunk 5]
```diff
@@ -46,7 +46,10 @@ final class DirectionVC: UIViewController {
         }
     }
     
-    var directionSearchView: DirectionSearchView = DirectionSearchView()
+    lazy var directionSearchView: DirectionSearchView = {
+        let titleTopOffset: CGFloat = isInSplitViewController ? 0 : 20
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Controller/DirectionVC.swift:50]: move to constants

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Views/RouteOptionsView.swift [hunk 6]
```diff
@@ -87,7 +87,7 @@ final class RouteOptionsView: UIView {
         
         containerView.isHidden = !routeOptionState
         routeOptionImage.image = UIImage(systemName: routeOptionState ? "chevron.up" : "chevron.down")
-        changeRouteOptionHeight?(routeOptionState ? 203 : 32)
+        changeRouteOptionHeight?(routeOptionState ? 144 : 32)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Views/RouteOptionsView.swift:90]: move to constants

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/POICard/Controller/POICardVC.swift [hunk 7]
```diff
@@ -29,7 +29,10 @@ final class POICardVC: UIViewController, UIViewControllerTransitioningDelegate {
         return locationManager
     }()
     
-    private let poiCardView: POICardView = POICardView()
+    private lazy var poiCardView: POICardView = {
+        let titleTopOffset: CGFloat = isInSplitViewController ? 0 : 20
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/POICard/Controller/POICardVC.swift:33]: move to constants

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Search/Controller/SearchVC.swift [hunk 8]
```diff
@@ -89,7 +94,11 @@ final class SearchVC: UIViewController {
         
         searchBarView.snp.makeConstraints {
             $0.top.equalTo(view.safeAreaLayoutGuide)
-            $0.height.equalTo(76)
+            if isInSplitViewController {
+                $0.height.equalTo(40)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Search/Controller/SearchVC.swift:98]: move to constants

File: LocationServices/LocationServices/Scenes/Explore/Views/ExploreView.swift [hunk 9]
```diff
@@ -708,7 +706,11 @@ private extension ExploreView {
         }
         
         amazonMapLogo.snp.makeConstraints {
-            $0.leading.equalToSuperview().offset(8)
+            if isiPad {
+                $0.bottom.equalTo(safeAreaLayoutGuide).offset(-8)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Explore/Views/ExploreView.swift:710]: move to constants

File: LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift [hunk 10]
```diff
@@ -89,7 +90,11 @@ final class GeofenceMapView: UIView {
         
         amazonMapLogo.snp.makeConstraints {
             $0.leading.equalToSuperview().offset(8)
-            $0.bottom.equalTo(searchBarView.snp.top).offset(-8)
+            if isiPad {
+                $0.bottom.equalTo(safeAreaLayoutGuide).offset(-8)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift:94]: move to constants

File: LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift [hunk 11]
```diff
@@ -102,7 +107,11 @@ final class GeofenceMapView: UIView {
         
         mapLayer.snp.makeConstraints {
             $0.top.trailing.equalToSuperview()
-            $0.bottom.equalTo(searchBarView.snp.top).offset(-16)
+            if isiPad {
+                $0.bottom.equalTo(safeAreaLayoutGuide).offset(8)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift:111]: move to constants

File: LocationServices/LocationServices/Scenes/Settings/Controller/SettingsVC.swift [hunk 12]
```diff
@@ -71,10 +71,9 @@ final class SettingsVC: UIViewController {
         self.view.addSubview(tableView)
         
         headerTitle.snp.makeConstraints {
-            $0.top.equalTo(self.view.safeAreaLayoutGuide).offset(16)
-            $0.leading.equalToSuperview().offset(24)
+            $0.top.equalTo(self.view.safeAreaLayoutGuide)
+            $0.leading.equalToSuperview().offset(16)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Settings/Controller/SettingsVC.swift:75]: move to constants

File: LocationServices/LocationServices/Scenes/Settings/Subviews/Data Provider/Controller/DataProviderVC.swift [hunk 13]
```diff
@@ -56,9 +55,9 @@ final class DataProviderVC: UIViewController {
         let isPad = UIDevice.current.userInterfaceIdiom == .pad
         if isPad {
             view.addSubview(screenTitleLabel)
-            screenTitleLabel.snp.makeConstraints { make in
-                make.top.equalTo(view.safeAreaLayoutGuide).offset(24)
-                make.horizontalEdges.equalToSuperview().inset(24)
+            screenTitleLabel.snp.makeConstraints {
+                $0.top.equalTo(view.safeAreaLayoutGuide)
+                $0.horizontalEdges.equalToSuperview().inset(16)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Settings/Subviews/Data Provider/Controller/DataProviderVC.swift:60]: move to constants

File: LocationServices/LocationServices/Scenes/Settings/Subviews/MapStyle/Controller/MapStyleVC.swift [hunk 14]
```diff
@@ -115,9 +114,9 @@ final class MapStyleVC: UIViewController {
         let isPad = UIDevice.current.userInterfaceIdiom == .pad
         if isPad {
             view.addSubview(screenTitleLabel)
-            screenTitleLabel.snp.makeConstraints { make in
-                make.top.equalTo(view.safeAreaLayoutGuide).offset(24)
-                make.horizontalEdges.equalToSuperview().inset(24)
+            screenTitleLabel.snp.makeConstraints {
+                $0.top.equalTo(view.safeAreaLayoutGuide)
+                $0.horizontalEdges.equalToSuperview().inset(16)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Settings/Subviews/MapStyle/Controller/MapStyleVC.swift:119]: move to constants

File: LocationServices/LocationServices/Scenes/Settings/Subviews/RouteOptions/Controller/RouteOptionVC.swift [hunk 15]
```diff
@@ -53,9 +52,9 @@ final class RouteOptionVC: UIViewController {
         let isPad = UIDevice.current.userInterfaceIdiom == .pad
         if isPad {
             view.addSubview(screenTitleLabel)
-            screenTitleLabel.snp.makeConstraints { make in
-                make.top.equalTo(view.safeAreaLayoutGuide).offset(24)
-                make.horizontalEdges.equalToSuperview().inset(24)
+            screenTitleLabel.snp.makeConstraints {
+                $0.top.equalTo(view.safeAreaLayoutGuide)
+                $0.horizontalEdges.equalToSuperview().inset(16)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Settings/Subviews/RouteOptions/Controller/RouteOptionVC.swift:57]: move to constants

File: LocationServices/LocationServices/Scenes/SideBar/Controller/SideBarVC.swift [hunk 16]
```diff
@@ -18,19 +28,24 @@ final class SideBarVC: UIViewController {
...
         setupTableView()
     }
     
     private func setupViews() {
+        view.addSubview(titleLabel)
         view.addSubview(tableView)
         
+        titleLabel.snp.makeConstraints {
+            $0.top.equalTo(view.safeAreaLayoutGuide)
+            $0.leading.equalTo(view.safeAreaLayoutGuide).offset(16)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/SideBar/Controller/SideBarVC.swift:42]: move to constants

File: LocationServices/LocationServices/Scenes/Tracking/Controller/TrackingVC.swift [hunk 17]
```diff
@@ -29,7 +29,7 @@ final class TrackingVC: UIViewController {
     }()
     
     private lazy var historyHeaderView: TrackingHistoryHeaderView = {
-        let view = TrackingHistoryHeaderView()
+        let view = TrackingHistoryHeaderView(titleTopOffset: 27)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Tracking/Controller/TrackingVC.swift:32]: move to constantsmove to constants

File: LocationServices/LocationServices/Scenes/Tracking/Subviews/Tracking History/Controller/TrackingHistoryVC.swift [hunk 18]
```diff
@@ -10,7 +10,11 @@ import SnapKit
 
 final class TrackingHistoryVC: UIViewController {
     
-    private(set) var headerView = TrackingHistoryHeaderView()
+    private var isiPad = UIDevice.current.userInterfaceIdiom == .pad
+    private(set) lazy var headerView: TrackingHistoryHeaderView = {
+        let titleTopOffset: CGFloat = isiPad ? 0 : 27
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Tracking/Subviews/Tracking History/Controller/TrackingHistoryVC.swift:15]: move to constants

File: LocationServices/LocationServices/Scenes/Tracking/Views/TrackingViews.swift [hunk 19]
```diff
@@ -85,12 +86,17 @@ final class TrackingMapView: UIView {
         
         mapLayer.snp.makeConstraints {
             $0.top.trailing.equalToSuperview()
-            $0.bottom.equalToSuperview().offset(-16)
+            if isiPad {
+                $0.bottom.equalTo(safeAreaLayoutGuide).offset(8)
```
Comment (vadymmakeendev) [inline LocationServices/LocationServices/Scenes/Tracking/Views/TrackingViews.swift:90]: move to constants

**Review by vadymmakeendev — CHANGES_REQUESTED** [review]

File: LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift (hunk 1 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift (hunk 2 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/About/Controller/AboutVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/About/Sub-Scenes/Attribution/Controller/AttributionVC.swift (hunk 3 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/About/Sub-Scenes/Attribution/Controller/AttributionVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Explore/Controller/ExploreVC.swift (hunk 4 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Explore/Controller/ExploreVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Controller/DirectionVC.swift (hunk 5 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Controller/DirectionVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Views/RouteOptionsView.swift (hunk 6 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Direction/Views/RouteOptionsView.swift]: fixed

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/POICard/Controller/POICardVC.swift (hunk 7 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/POICard/Controller/POICardVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Search/Controller/SearchVC.swift (hunk 8 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Explore/Sub-Scenes/Search/Controller/SearchVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Explore/Views/ExploreView.swift (hunk 9 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Explore/Views/ExploreView.swift]: fixed

File: LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift (hunk 10 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift]: fixed

File: LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift (hunk 11 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Geofence/Views/GeofenceView.swift]: fixed

File: LocationServices/LocationServices/Scenes/Settings/Controller/SettingsVC.swift (hunk 12 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Settings/Controller/SettingsVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Settings/Subviews/Data Provider/Controller/DataProviderVC.swift (hunk 13 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Settings/Subviews/Data Provider/Controller/DataProviderVC.swift:57]: fixed

File: LocationServices/LocationServices/Scenes/Settings/Subviews/MapStyle/Controller/MapStyleVC.swift (hunk 14 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Settings/Subviews/MapStyle/Controller/MapStyleVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Settings/Subviews/RouteOptions/Controller/RouteOptionVC.swift (hunk 15 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Settings/Subviews/RouteOptions/Controller/RouteOptionVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/SideBar/Controller/SideBarVC.swift (hunk 16 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/SideBar/Controller/SideBarVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Tracking/Controller/TrackingVC.swift (hunk 17 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Tracking/Controller/TrackingVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Tracking/Controller/TrackingVC.swift [hunk 20]
```diff
@@ -186,6 +187,19 @@ final class TrackingVC: UIViewController {
         grabberIcon.isHidden = isVisible
     }
     
+    @objc private func authorizationStatusChanged(_ notification: Notification) {
+        DispatchQueue.main.async {
+            switch LoginViewModel.getAuthStatus() {
+            case .authorized:
+                self.trackingMapView.adjustMapLayerItems(bottomSpace: 70)
```
[the remark this answers is not in the document]
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Tracking/Controller/TrackingVC.swift]: fixed

File: LocationServices/LocationServices/Scenes/Tracking/Views/TrackingViews.swift (hunk 19 above)
Comment (denis-ryzhkov-makeen) [reply LocationServices/LocationServices/Scenes/Tracking/Views/TrackingViews.swift]: fixed

**Review by vadymmakeendev — APPROVED** [review]

**Pull request merged**